### PR TITLE
v4.0 changelog

### PR DESCRIPTION
**Pull request by JeffSchering**

### CODE REVIEW

**Comment by jasonblais** [comment]
Ready for review now

**Comment by JeffSchering** [comment]
@jasonblais All the ones that I'm aware of seem to be there.

File: source/administration/changelog.md [hunk 1]
```diff
@@ -4,9 +4,269 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+
+### Highlights
+
+#### Native iOS and Android Apps
+- Second generation mobile apps released for iOS and Android.
+- The apps are [EMM compatible starting with BlackBerry Dynamics](https://about.mattermost.com/mattermost-2nd-gen-mobile-apps-released-emm-compatible-starting-with-blackberry-dynamics/).
+
+#### Updated Web UI
+- Updated the look of channel header and channel sidebar in the web user interface.
+- Added a new default theme, "Mattermost". To try it, go to **Account Settings > Display > Theme**.
```
Comment (esethna) [inline source/administration/changelog.md]: Didn't we always have a theme "Mattermost"? We just updated the vector for it I thought?

File: source/administration/changelog.md [hunk 2]
```diff
@@ -4,9 +4,269 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+
+- Mattermost v4.0.0 contains multiple security fixes. [Upgrading](http://docs.mattermost.com/administration/upgrade.html) is highly recommended. Details will be posted on our [security updates page](https://about.mattermost.com/security-updates/) 14 days after release as per the [Mattermost Responsible Disclosure Policy](https://www.mattermost.org/responsible-disclosure-policy/).
+
+### Highlights
+
+#### Native iOS and Android Apps
+- Second generation mobile apps released for iOS and Android.
+- The apps are [EMM compatible starting with BlackBerry Dynamics](https://about.mattermost.com/mattermost-2nd-gen-mobile-apps-released-emm-compatible-starting-with-blackberry-dynamics/).
+
+#### Updated Web UI
```
Comment (esethna) [inline source/administration/changelog.md]: UI > User Interface (to be consistent in other places in the changelog)

File: source/administration/changelog.md [hunk 3]
```diff
@@ -4,9 +4,269 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+- Added support for experimental gossip protocol, where the server will attempt to communicate via the gossip protocol over the gossip port.
+
+### Improvements
+
+#### Web User Interface
+- Adjusted post spacing to be consistent across Markdown formatting, replies and consecutive posts.
+- On hover colour for pin and channel member icons now consistent with flag and recent mentions icons.
+- Emojis are now vertically aligned in post view.
+- Channel name, header and purpose now update real time for all users.
+- For reply threads in the center channel, the "Commented on" phrase now respects the teammate name display setting.
```
Comment (esethna) [inline source/administration/changelog.md]: Wasn't that setting removed?

File: source/administration/changelog.md [hunk 4]
```diff
@@ -4,9 +4,269 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+- If a message starts with a / but fails to send (either due to timeout or invalid command), the message is put back to the input box.
+
+#### Bulk Import Tool
+- Added support for direct message channels and posts to the [bulk import tool](https://docs.mattermost.com/deployment/bulk-loading.html).
+
+#### Authentication
+- User creation via OAuth (GitLab/Google/Office365) properly restricted to accepted domains, [if specified](https://docs.mattermost.com/administration/config-settings.html#restrict-account-creation-to-specified-email-domains).
+- **Invite New Member** dialog validates email address against accepted domains, if set.
+
+#### New URL Routes // XXX Need better name and description of benefit
```
Comment (esethna) [inline source/administration/changelog.md]: Can this comment be removed?

Comment (esethna) [inline source/administration/changelog.md]: Also should this go in the API section?

File: source/administration/changelog.md [hunk 5]
```diff
@@ -4,9 +4,269 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+- An empty push notification no longer sent for messages only containing file attachments.
+- Custom emoji search results no longer filter by creator's first and last name.
+- `/expand` and `/collapse` slash commands now properly collapse images in website link previews.
+- Group Message channels that are favorited can now be closed.
+- Deactivated users now properly listed in Direct and Group Message channels in the left-hand sidebar.
+- Fixed search in team and channel Manage Members dialog.
+- File upload cancelled if you click "x" on thumbnail while file is uploading in your message draft.
+- Status no longer appears offline after joining a new team.
+- An empty push notification is no longer sent for messages only containing file attachments.
+- If you're at the top of a channel when a new post is received, the view no longer shifts and removes the oldest post that was in your view.
```
Comment (esethna) [inline source/administration/changelog.md]: Not really sure what this means. Is it that the center channel maintains scroll postition when new messages are received in the channel?

File: source/administration/changelog.md [hunk 6]
```diff
@@ -4,9 +4,269 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+- If a message starts with a / but fails to send (either due to timeout or invalid command), the message is put back to the input box.
+
+#### Bulk Import Tool
+- Added support for Direct Message channels and posts to the [bulk import tool](https://docs.mattermost.com/deployment/bulk-loading.html).
+
+#### Authentication
+- User creation via OAuth (GitLab/Google/Office365) properly restricted to accepted domains, [if specified](https://docs.mattermost.com/administration/config-settings.html#restrict-account-creation-to-specified-email-domains).
+- **Invite New Member** dialog validates email addresses against accepted domains, if set.
+
+#### New URL Routes // XXX Need better name and description of benefit
```
Comment (esethna) [inline source/administration/changelog.md]: Should this be moved to the API section? Can the comment be removed?

File: source/administration/changelog.md [hunk 7]
```diff
@@ -4,9 +4,269 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+
+### Highlights
+
+#### Native iOS and Android Apps
+- Second generation mobile apps released for [iOS](https://itunes.apple.com/us/app/mattermost/id1257222717?mt=8) and [Android](https://play.google.com/store/apps/details?id=com.mattermost.rn).
+- The apps are [EMM compatible starting with BlackBerry Dynamics](https://about.mattermost.com/mattermost-2nd-gen-mobile-apps-released-emm-compatible-starting-with-blackberry-dynamics/).
+
+#### Updated Web User Interface
+- Updated the appearance of channel header and channel sidebar in the web user interface.
+- Added a new default theme, "Mattermost". To try it, go to **Account Settings > Display > Theme**.
```
Comment (esethna) [inline source/administration/changelog.md]: Didn't we always have a theme "Mattermost"? We just updated the vector for it I thought?

File: source/administration/changelog.md [hunk 8]
```diff
@@ -4,9 +4,269 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
+- Added support for experimental gossip protocol, where the server will attempt to communicate via the gossip protocol over the gossip port.
+
+### Improvements
+
+#### Web User Interface
+- Adjusted post spacing to be consistent across Markdown formatting, replies and consecutive posts.
+- On hover colour for pin and channel member icons now consistent with flag and recent mentions icons.
+- Emojis are now vertically aligned in post view.
+- Channel name, header and purpose now update in real time for all users.
+- For reply threads in the center channel, the "Commented on" phrase now respects the teammate name display setting.
```
Comment (esethna) [inline source/administration/changelog.md]: Wasn't that setting moved to the system console?

File: source/administration/changelog.md [hunk 9]
```diff
@@ -4,9 +4,269 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
...
 
-## Release v3.10.0
+## Release v4.0.0
 
-Release Date: June 16, 2017
+Release Date: July 16, 2017
+
+### Security Update
+
+- Mattermost v4.0.0 contains multiple security fixes. [Upgrading](http://docs.mattermost.com/administration/upgrade.html) is highly recommended. Details will be posted on our [security updates page](https://about.mattermost.com/security-updates/) 14 days after release as per the [Mattermost Responsible Disclosure Policy](https://www.mattermost.org/responsible-disclosure-policy/).
```
Comment (jasonblais) [inline source/administration/changelog.md]: Yes, added

File: source/administration/changelog.md (hunk 6 above)
Comment (esethna) [inline source/administration/changelog.md]: Can this comment be removed? Also should this be moved to the API section?

File: source/administration/changelog.md (hunk 5 above)
Comment (jasonblais) [inline source/administration/changelog.md]: Yes, that's better :) Will update

**Comment by esethna** [comment]
Thanks @jasonblais, committed some tweaks and added some other comments above ^

File: source/administration/changelog.md (hunk 7 above)
Comment (jasonblais) [inline source/administration/changelog.md]: So we have a "new default theme". The name per say didn't change

0/5 if we want to change the wording

File: source/administration/changelog.md (hunk 8 above)
Comment (jasonblais) [inline source/administration/changelog.md]: Yes, but it still respects it. I'll change to `teammate name display config setting`

File: source/administration/changelog.md (hunk 6 above)
Comment (jasonblais) [inline source/administration/changelog.md]: Whoops yes. They're not API changes, they're URL changes.

Like you can open a DM channel with someone by pasting a URL of the form https://pre-release.mattermost.com/messages/@jason

**Comment by jasonblais** [comment]
Thanks @esethna! Updated

File: source/administration/changelog.md (hunk 7 above)
Comment (esethna) [inline source/administration/changelog.md]: I changed to "Updated"

File: source/administration/changelog.md (hunk 6 above)
Comment (esethna) [inline source/administration/changelog.md]: Ah got it, cool

**Pull request merged**